Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrExtensions.kt

```diff
@@ -7,14 +7,30 @@ package org.jetbrains.kotlin.native.interop.gen
 import org.jetbrains.kotlin.native.interop.indexer.ObjCProtocol
 
 private val StubOrigin.ObjCMethod.isOptional: Boolean
-        get() = container is ObjCProtocol && method.isOptional
+    get() = container is ObjCProtocol && method.isOptional
 
-fun FunctionStub.isObjCMethodOptional(): Boolean = this.origin is StubOrigin.ObjCMethod &&
+fun FunctionStub.isOptionalObjCMethod(): Boolean = this.origin is StubOrigin.ObjCMethod &&
         this.origin.isOptional
 
 val StubContainer.isInterface: Boolean
-        get() = if (this is ClassStub.Simple) {
-                modality == ClassStubModality.INTERFACE
-        } else {
-                false
-        }
\ No newline at end of file
+    get() = if (this is ClassStub.Simple) {
+        modality == ClassStubModality.INTERFACE
+    } else {
+        false
+    }
+
+/**
+ * Compute which names will be declared by [StubContainer] in the given [pkgName]
+ */
+fun StubContainer.computeNamesToBeDeclared(pkgName: String): List<String> {
+
+    fun List<Classifier>.topLevelNamesInPackage() =
+            filter { it.pkg == pkgName }.map { it.topLevelName }
+
+    val classNames = classes.filterIsInstance<ClassStub.Simple>().map { it.classifier }.topLevelNamesInPackage()
+    val enumNames = classes.filterIsInstance<ClassStub.Enum>().map { it.classifier }.topLevelNamesInPackage()
```

Comment:
This code would be more reliable if it was using `when` over sealed class instead of multiple `filterIsInstance`.